sql/common/V2.62.0__user_saved_regions.sql

```diff
@@ -0,0 +1,103 @@
+-- Create table for storing user-defined regions
+-- Users can save frequently used custom regions for reuse in downloads and searches
+
+CREATE TABLE IF NOT EXISTS user_region (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    sub UUID NOT NULL,
+    name VARCHAR(255) NOT NULL,
+    description TEXT,
+    geojson TEXT NOT NULL, -- Store the GeoJSON as text
+    geometry geometry(Geometry, 4326) GENERATED ALWAYS AS (ST_GeomFromGeoJSON(geojson)) STORED, -- Auto-generate PostGIS geometry
+    bbox FLOAT[] GENERATED ALWAYS AS (
+        ARRAY[
+            ST_XMin(ST_GeomFromGeoJSON(geojson))::FLOAT,
+            ST_YMin(ST_GeomFromGeoJSON(geojson))::FLOAT,
+            ST_XMax(ST_GeomFromGeoJSON(geojson))::FLOAT,
+            ST_YMax(ST_GeomFromGeoJSON(geojson))::FLOAT
+        ]
+    ) STORED, -- Auto-calculate bounding box in WGS84
+    area_sqkm FLOAT GENERATED ALWAYS AS (
+        ST_Area(ST_Transform(ST_GeomFromGeoJSON(geojson), 5070)) / 1000000.0
+    ) STORED, -- Auto-calculate area in square kilometers
+    created_at TIMESTAMPTZ DEFAULT NOW(),
+    updated_at TIMESTAMPTZ DEFAULT NOW(),
+    is_public BOOLEAN DEFAULT FALSE, -- Allow sharing regions with other users
+    tags TEXT[], -- Optional tags for categorization
+    CONSTRAINT unique_user_region_name UNIQUE(sub, name) -- Each user must have unique region names
+);
+
+-- Index for fast lookups
+CREATE INDEX idx_user_region_sub ON user_region(sub);
+CREATE INDEX idx_user_region_geometry ON user_region USING GIST(geometry);
+CREATE INDEX idx_user_region_public ON user_region(is_public) WHERE is_public = TRUE;
+CREATE INDEX idx_user_region_tags ON user_region USING GIN(tags);
+
+-- Function to validate GeoJSON before saving
+CREATE OR REPLACE FUNCTION validate_user_region_geojson()
```

Comment:
Create a temporary empty trigger function here and move the actual trigger to a repeatable migration.

Makes it easier to update the trigger itself later.